libs/sdk-backend-base/src/cachingBackend/index.ts

```diff
@@ -380,8 +455,52 @@ export type CachingConfiguration = {
      * @param cacheControl - cache control instance
      */
     onCacheReady?: (cacheControl: CacheControl) => void;
+
+    /**
+     * Maximum number of organizations that will have its security settings cached. The scope string built by
+     * particular backend implementation from organization ID will be used as cache key.
+     *
+     * When limit is reached, cache entries will be evicted using LRU policy.
+     *
+     * When no maximum number is specified, the cache will be unbounded and no evictions will happen. Unbounded
+     * security settings organization cache is dangerous in applications with long-lived sessions that can
+     * create many unique requests and memory usage will only go up.
+     *
+     * When non-positive number is specified, then no caching will be done.
+     */
+    maxSecuritySettingsOrganizations: number | undefined;
+
+    /**
+     * Maximum number of URLs per organization that will have its validation result cached. The URL
+     * and validation context is used to form a cache key.
+     *
+     * When limit is reached, cache entries will be evicted using LRU policy.
+     *
+     * When no maximum number is specified, the cache will be unbounded and no evictions will happen. Unbounded
+     * security settings organization URL cache is dangerous in applications with long-lived sessions that can
+     * create many unique requests and memory usage will only go up.
+     *
+     * Setting non-positive number here is invalid. If you want to turn off organization security settings caching, tweak the `maxSecuritySettingsOrganizations`.
+     */
+    maxSecuritySettingsOrganizationUrls: number | undefined;
+
+    /**
+     * Maximum age of cached organization's URL validation results. The value is in milliseconds.
+     *
+     * Items are not pro-actively pruned out as they age, but if you try to get an item that is too old, it'll drop it and return undefined instead of giving it to you. The purpose of the cache setting is not mainly to limit its size to prevent memory usage grow (tweak `maxSecuritySettingsOrganizationUrls` for that) but to propagate URL whitelist changes on backend to the long-lived application sessions.
```

Comment:
is the part about 'return undefined' valid? the underlying cache will do that, however in that case our logic will re-validate and user still gets true/false. the way how this is worded sounds like you may suddenly get `undefined` as result of validation call

Reply:
You are right. Not sure what I was thinking about when I was putting it together 😅